game.py

```diff
@@ -20,8 +20,35 @@ def snowman(snowman_word):
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
-
+    # Initialize the correct_letter_guess_statuses dictionary
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_list = []
+
+    # Loop until the player wins or loses
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES:
+        print("\nWord: " + generate_word_progress_string(snowman_word, correct_letter_guess_statuses))
+        print("Wrong guesses:", " ".join(sorted(wrong_guesses_list)))
+        # Get a letter from the user
+        user_input = get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list)
+        # Check if the user input is in the word
+        if user_input in correct_letter_guess_statuses:
+            print("This letter is in the word!")
+            correct_letter_guess_statuses[user_input] = True
+        # If the user input is not in the word
+        else:
+            print(f"This letter {user_input} is not in the word")
+            wrong_guesses_list.append(user_input)
+        # Print the snowman graphic
+        print_snowman_graphic(len(wrong_guesses_list))
+
+        # If the player wins
+        if is_word_guessed(snowman_word, correct_letter_guess_statuses):
+            print("Congratulations, you win!")
+            return
```

Comment:
Good call to check for the win condition within the loop and return the function early if the win condition is met! This is a very popular and widely used structure!